aws/structure.go

```diff
@@ -1045,6 +1045,32 @@ func expandESEBSOptions(m map[string]interface{}) *elasticsearch.EBSOptions {
 	return &options
 }
 
+func flattenESEncryptAtRestOptions(o *elasticsearch.EncryptionAtRestOptions) []map[string]interface{} {
+	m := map[string]interface{}{}
```

Comment:
Not sure I follow that, what would cause the crash? If `o` is nil doesn't it already return an empty `[]map[string]interface{}{}` due to the nil checks on `o.Enabled` and `o.KmsKeyId`?

Reply:
> If o is nil doesn't it already return an empty []map[string]interface{}{} due to the nil checks on o.Enabled and o.KmsKeyId?

No, it will crash on line 1052 in such case because that condition is trying to access a field of a struct which isn't struct, but `nil`. In other words that condition is one step ahead.

See https://play.golang.org/p/R07zPiMT92M

Reply:
Right, sorry, completely missed that for some reason. Fixed now, thanks for the review.

Reply:
No worries 😸  That's why we do reviews.